packages/rest/src/controllers/connections/ConnectionController.ts

```diff
@@ -0,0 +1,188 @@
+    try {
+      const connection = await this.agent.connections.acceptInvitation(connectionId);
+
+      return this.mapConnection(connection);
+    } catch (error) {
+      if (error instanceof RecordNotFoundError) {
+        throw new NotFoundError(`connection with connectionId "${connectionId}" not found.`);
+      }
+      throw new InternalServerError(`something went completely wrong: ${error}`);
+    }
+  }
+
+  /**
+   * Accept a connection request as inviter (by sending a connection response message) for the connection with the specified connection id.
+   * This is not needed when auto accepting of connection is enabled.
+   */
+  @Post("/:connectionId/accept-request")
+  public async acceptRequest(@Param("connectionId") connectionId: string): Promise<ConnectionModel> {
+    try {
+      const connection = await this.agent.connections.acceptRequest(connectionId);
+
+      return this.mapConnection(connection);
+    } catch (error) {
+      if (error instanceof RecordNotFoundError) {
+        throw new NotFoundError(`connection with connectionId "${connectionId}" not found.`);
+      }
+      throw new InternalServerError(`something went completely wrong: ${error}`);
+    }
+  }
+
+  /**
+   * Accept a connection response as invitee (by sending a trust ping message) for the connection with the specified connection id.
+   * This is not needed when auto accepting of connection is enabled.
+   */
+  @Post("/:connectionId/accept-response")
+  public async acceptResponse(@Param("connectionId") connectionId: string): Promise<ConnectionModel> {
+    try {
+      const connection = await this.agent.connections.acceptResponse(connectionId);
+
+      return this.mapConnection(connection);
+    } catch (error) {
+      if (error instanceof RecordNotFoundError) {
+        throw new NotFoundError(`connection with connectionId "${connectionId}" not found.`);
+      }
+      throw new InternalServerError(`something went completely wrong: ${error}`);
+    }
+  }
+
+  /**
+   * Deletes a connectionRecord in the connectionRepository.
+   */
+  @Delete("/:connectionId")
+  public async deleteConnection(@Param("connectionId") connectionId: string): Promise<ConnectionModel> {
+    try {
+      const connectionRepository = this.agent.injectionContainer.resolve(ConnectionRepository);
+      const connection = await connectionRepository.getById(connectionId);
+      await connectionRepository.delete(connection);
+
+      return this.mapConnection(connection);
+    } catch (error) {
+      if (error instanceof RecordNotFoundError) {
+        throw new NotFoundError(`connection with connectionId "${connectionId}" not found.`);
+      }
+      throw new InternalServerError(`something went completely wrong: ${error}`);
+    }
+  }
+
+  private mapConnection(connection: ConnectionRecord): ConnectionModel {
+    return {
+      state: connection.state,
+      connection_id: connection.id,
+      connection,
+    };
```

Comment:
Although easier when using swagger UI, I'd rather directly return the connection record. The state and connection_id properties are also present in the connection object